pkg/providers/vsphere/controlplane_test.go

```diff
@@ -105,47 +117,56 @@ func TestControlPlaneObjects(t *testing.T) {
 	}
 }
 
-func capiCluster() *clusterv1.Cluster {
-	return &clusterv1.Cluster{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "Cluster",
-			APIVersion: "cluster.x-k8s.io/v1beta1",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "cluster",
-			Namespace: constants.EksaSystemNamespace,
-		},
-		Spec: clusterv1.ClusterSpec{
-			InfrastructureRef: &corev1.ObjectReference{
-				Name:       "cluster",
-				Namespace:  constants.EksaSystemNamespace,
-				Kind:       "VSphereCluster",
-				APIVersion: "infrastructure.cluster.x-k8s.io/v1beta1",
-			},
-			ControlPlaneRef: &corev1.ObjectReference{
-				Name:       "cp",
-				Namespace:  constants.EksaSystemNamespace,
-				Kind:       "KubeadmControlPlane",
-				APIVersion: "controlplane.clusterapi.k8s/v1beta1",
-			},
-		},
-	}
+func TestControlPlaneSpecNewCluster(t *testing.T) {
+	g := NewWithT(t)
+	logger := test.NewNullLogger()
+	ctx := context.Background()
+	client := test.NewFakeKubeClient()
+	spec := givenClusterSpec(t, testClusterConfigMainFilename)
+
+	cp, err := vsphere.ControlPlaneSpec(ctx, logger, client, spec)
+	g.Expect(err).NotTo(HaveOccurred())
+	g.Expect(cp).NotTo(BeNil())
+	g.Expect(cp.Cluster).To(Equal(capiCluster()))
+	g.Expect(cp.KubeadmControlPlane).NotTo(BeNil())
+	g.Expect(cp.EtcdCluster).NotTo(BeNil())
+	g.Expect(cp.ProviderCluster).NotTo(BeNil())
+	g.Expect(cp.ControlPlaneMachineTemplate.Name).To(Equal("test-control-plane-1"))
+	g.Expect(cp.EtcdMachineTemplate.Name).To(Equal("test-etcd-1"))
+	g.Expect(cp.Secrets).NotTo(BeNil())
+	g.Expect(cp.ConfigMaps).NotTo(BeNil())
```

Comment:
Can we add `Equal` expectations for all objects in the same way we have it for `Cluster`?